test/battle/move_effect/embargo.c

```diff
@@ -0,0 +1,394 @@
+#include "global.h"
+#include "test/battle.h"
```

Comment:
I would add an Assumption before the tests, that `gBattleMoves[MOVE_EMBARGO].effect == EFFECT_EMBARGO`